### ISSUE TITLE
🐛 Using `extends` disables support for `.gitignore`

### ISSUE THREAD

**Issue by SlexAxton** [biomejs/biome#5439]
### Environment information

```block
CLI:
  Version:                      1.9.4
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           "v22.14.0"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "pnpm/10.6.5"

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    false
  VCS disabled:                 false

Workspace:
  Open Documents:               0
```

### What happened?

If I have a root `biome.jsonc` file that looks like this:

```jsonc
{
	"$schema": "https://biomejs.dev/schemas/1.9.4/schema.json",
	"vcs": { "enabled": true, "clientKind": "git", "useIgnoreFile": true },
	"linter": {
		"enabled": true,
		"rules": {
			"recommended": true
		}
	}
}
```

And I have a `.gitignore` file that has:

```
*.ignore.ts
```

And then I have a `good.ts` file:

```ts
export const good = "all good";
```

And then I have a `bad.ignore.ts` file:

```ts
const foo = `fails a rule`;
```

And I run:

```sh
biome lint
```

Then it successfully ignores the `bad.ignore.ts` file.

But if I create `biome.more.jsonc` with:

```jsonc
{
	"$schema": "https://biomejs.dev/schemas/1.9.4/schema.json",
	"extends": "./biome.jsonc"
}
```

And then I run:

```sh
biome lint --config-path=./biome.more.jsonc
```

Then I get a lint error in `bad.ignore.ts`.

I have tried also adding the same `vcs` config to `biome.more.jsonc` but to no change.

### Expected result

I'd expect both files to treat `.gitignore` the same, and for there to be no reported errors.

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct
